src/Player.tsx

```diff
@@ -6,311 +6,277 @@ import { popNote } from './music-sheet';
+            'A#4': 'As4.[mp3|ogg]',
+            'B3': 'B3.[mp3|ogg]',
+            'B4': 'B4.[mp3|ogg]',
+            'C4': 'C4.[mp3|ogg]',
+            'C5': 'C5.[mp3|ogg]',
+            'C#3': 'Cs3.[mp3|ogg]',
+            'C#4': 'Cs4.[mp3|ogg]',
+            'C#5': 'Cs5.[mp3|ogg]',
+            'D3': 'D3.[mp3|ogg]',
+            'D4': 'D4.[mp3|ogg]',
+            'D5': 'D5.[mp3|ogg]',
+            'D#3': 'Ds3.[mp3|ogg]',
+            'D#4': 'Ds4.[mp3|ogg]'
+        },
+        release: 1,
+        baseUrl: "https://coleim.github.io/tonejs-instruments/samples/saxophone/",
+        onload
+    }).toDestination();
+}
 
-    const minorChordsAssociation = [
-        ["C3", "Eb3", "G3"],   // Do (C), Mi♭ (E♭), Sol (G)
-        ["D3", "F3", "A3"],   // Ré (D), Fa (F), La (A)
-        ["E3", "G3", "B3"],   // Mi (E), Sol (G), Si (B)
-        ["F3", "Ab3", "C3"],  // Fa (F), La♭ (A♭), Do (C)
-        ["G3", "Bb3", "D3"],   // Sol (G), Si♭ (B♭), Ré (D)
-        ["A3", "C3", "E3"],   // La (A), Do (C), Mi (E)
-        ["B3", "D3", "F#3"],   // Si (B), Ré (D), Fa♯ (F♯)
-    ];
+function createXylophone(onload: () => void) {
+    return new Tone.Sampler({
+        urls: {
+            'C8': 'C8.[mp3|ogg]',
+            'G4': 'G4.[mp3|ogg]',
+            'G5': 'G5.[mp3|ogg]',
+            'G6': 'G6.[mp3|ogg]',
+            'G7': 'G7.[mp3|ogg]',
+            'C5': 'C5.[mp3|ogg]',
+            'C6': 'C6.[mp3|ogg]',
+            'C7': 'C7.[mp3|ogg]'
+        },
+        release: 1,
+        baseUrl: "https://coleim.github.io/tonejs-instruments/samples/xylophone/",
+        onload
+    }).toDestination();
+}
 
-    // https://tonejs.github.io/docs/14.7.77/type/Subdivision
-    const notesDuration = [
-        "2n",
-        "4n",
-        "4n",
-        "4n",
-        "8n",
-        "8n",
-        "8n"
-    ];
 
-    function play() {
-        if(notes.length === 0) return;
-        if (!loop.current || loop.current.state === "stopped") {
-            loop.current = new Tone.Loop(function (time) {
-                if(!pausedRef.current) {
-                    if( headNote.current !== undefined) {
-                        let duration = "4n";
-                        if(nextNote.current !== undefined) {
-                            duration = notesDuration[nextNote.current];
-                        }
-                        currentInstrument.current.triggerAttackRelease(notesAssociation[headNote.current], duration, time);
-                        loop.current.interval = duration;
-                        popNote();
-                    } else {
-                        stop();
-                    }
-                }
-            }, "4n").start(0);
+const useInstrument = (id: string) => {
+    const [loading, setLoading] = useState(false)
+    const instrumentRef = useRef<ReturnType<typeof createInstrument>>()
+    useEffect(() => {
+        setLoading(true)
+        instrumentRef.current = createInstrument(id, () => setLoading(false))
```

Comment:
Changing the `isLoading` state here guarantees that the component is re-rendered as soon as the instrument ref is loaded.